cog/core/singleton.py

```
@@ -0,0 +1,8 @@
+class SingletonMeta(type):
+    _instances = {}
```

Ruff (0.8.2)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)